### PR TITLE
Update Crane to serve manifest schema version 2.

### PR DESCRIPTION
**Pull request by ipanova**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -263,6 +253,14 @@ Example ``apache.conf`` file:
 Release Notes
 -------------
 
+2.2.0
+
+The 3.0.0 release of Crane adds support for the Dcoker v2 API manifests schema2.
```
Comment (ipanova) [inline docs/index.rst]: will change the mistake to 2.2.0

File: crane/views/v2.py [hunk 2]
```diff
@@ -60,6 +61,17 @@ def name_redirect(relative_path):
     base_url = repository.get_path_for_repo(name_component)
     if not base_url.endswith('/'):
         base_url += '/'
+
+    if 'manifests' in path_component:
+        manifest, identifier = path_component.split('/')
+        tags_json, accept_headers = repository.get_tags_for_repo(name_component)
```
Comment (ipanova) [inline crane/views/v2.py]: good point, i will try that

**Comment by ipanova** [comment]
@dkliban 
pull of schema2
```
[ipanova@ina pulp_docker]$ docker pull localhost:5000/synctest:musl
musl: Pulling from synctest
36ad32b9a47a: Pull complete 
Digest: sha256:d2ac822f2e240c1718f99e283257e0500f02546085931358217790f53167510c
Status: Downloaded newer image for localhost:5000/synctest:musl
```
pull of schema1
```
[ipanova@ina pulp_docker]$ docker pull localhost:5000/synctest:ubuntu-12.04
ubuntu-12.04: Pulling from synctest
a3ed95caeb02: Pull complete 
205abe09039b: Pull complete 
Digest: sha256:9f44ce28d99752c237d942e865104adb86a2170822f5c6f5f028a44eb123b7d7
Status: Downloaded newer image for localhost:5000/synctest:ubuntu-12.04
[ipanova@ina pulp_docker]$ 
```

File: crane/views/v2.py [hunk 3]
```diff
@@ -60,10 +64,31 @@ def name_redirect(relative_path):
     base_url = repository.get_path_for_repo(name_component)
     if not base_url.endswith('/'):
         base_url += '/'
+
+    if 'manifests' in path_component:
```
Comment (ipanova) [inline crane/views/v2.py]: this logic is not correct, it should not make 302 redirects for docker pull by digest for schema2. I need to update this

**Comment by dkliban** [comment]
When performing a docker pull for a tag that does not exist, docker client experiences an error:

[root@dev conf.d]# docker pull localhost:5001/mydocker2:blah
Error response from daemon: error parsing HTTP 404 response body: invalid character '<' looking for beginning of value: "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n<html><head>\n<title>404 Not Found</title>\n</head><body>\n<h1>Not Found</h1>\n<p>The requested URL /pulp/docker/v2/mydocker2/manifests/blah was not found on this server.</p>\n</body></html>\n"

**Comment by dkliban** [comment]
When performing a docker pull for a digest that does not exist, docker client experiences an error:
`[root@dev conf.d]# docker pull localhost:5001/mydocker2@sha256:26947af801ea7f29ee3363b37c7f0e4110c5dc55989ca334b6d81e27be2a48cd`

`Error response from daemon: error parsing HTTP 404 response body: invalid character '<' looking for beginning of value: "<!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n<html><head>\n<title>404 Not Found</title>\n</head><body>\n<h1>Not Found</h1>\n<p>The requested URL /pulp/docker/v2/mydocker2/manifests/sha256:26947af801ea7f29ee3363b37c7f0e4110c5dc55989ca334b6d81e27be2a48cd was not found on this server.</p>\n</body></html>\n"`

**Comment by ipanova** [comment]
@dkliban that is ok, there is an upstream docker bug opened by @daviddavis

**Comment by dkliban** [comment]
@ipanova you are right. We also already have an issue in redmine tracking this problem https://pulp.plan.io/issues/2536

**Comment by ipanova** [comment]
ok test

File: docs/index.rst [hunk 4]
```diff
@@ -263,6 +253,14 @@ Example ``apache.conf`` file:
 Release Notes
 -------------
 
+2.2.0
+
+The 2.2.0 release of Crane adds support for the Dcoker v2 API manifests schema version 2.
+Crane is able to read new json data file format that includes knowledge of manifest schema data.
```
Comment (dkliban) [inline docs/index.rst]: I don't know if you need the next sentence here.

File: crane/views/v2.py [hunk 5]
```diff
@@ -60,6 +61,20 @@ def name_redirect(relative_path):
...
+        schema2_data_json = repository.get_schema2_data_for_repo(name_component)
+        if schema2_data_json is not None:
```
Comment (ipanova) [inline crane/views/v2.py]: i will change this, thanks

**Pull request merged**